rotkehlchen/chain/evm/decoding/gearbox/decoder.py

```diff
@@ -66,20 +66,36 @@ def pools(self) -> dict[ChecksumEvmAddress, GearboxPoolData]:
     def _cache_mapping_methods(self) -> tuple[Callable[[DecoderContext], DecodingOutput]]:
         return (self._decode_pool_events,)
 
-    def _decode_common(self, context: DecoderContext, user_address_bytes: bytes) -> tuple[ChecksumEvmAddress, GearboxPoolData, FVal, FVal] | None:  # noqa: E501
-        """This function decodes the common data for the deposit and withdraw events."""
-        if not self.base.is_tracked(user_address := hex_or_bytes_to_address(user_address_bytes)):
-            return None
+    def _is_weth_pool(self, token: EvmToken) -> bool:
+        """Check if one side of the pool includes WETH by verifying if the underlying token of the
+        farming token is WETH."""
+        return token.underlying_tokens[0].get_identifier(self.evm_inquirer.chain_id) in {
+            A_WETH,
+            A_WETH_ARB,
+            A_WETH_OPT,
+        }
 
+    def _decode_common(self, context: DecoderContext, address_offset: int | None = None) -> tuple[ChecksumEvmAddress, GearboxPoolData, FVal, FVal] | None:  # noqa: E501
+        """This function decodes the common data for the deposit and withdraw events."""
         try:
             pool_info = self.pools[context.tx_log.address]
         except KeyError:
             log.error(f'Could not find {self.evm_inquirer.chain_name} Gearbox pool info for {context.tx_log.address}')  # noqa: E501
             return None
 
+        token = EvmToken(pool_info.farming_pool_token)
+        if address_offset is None:
+            try:
+                address_offset = 0 if self._is_weth_pool(token) is True else 32
+            except IndexError:
+                return None
```

Comment:
do you need the argument then? If you have a rool to detect it why use the argument?

Reply:
@yabirgb it's for the _decode_withdraw. On redeem, it's always index 1, https://github.com/Gearbox-protocol/integrations-v3/blob/4bd6e3094812883464bbfa6467262123e3c72e50/contracts/interfaces/zappers/IZapper.sol